core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -15,104 +15,137 @@
  * stack trace for the throwable will be included in the log message.</p>
  *
  * <p>A minimum logging level threshold is determined by the {@link BaseConfigurations#AZURE_LOG_LEVEL AZURE_LOG_LEVEL}
  * environment configuration, by default logging is disabled. The default logging level for messages is
- * {@link ServiceLogger#asInfo() info}.</p>
+ * {@link ClientLogger#asInfo() info}.</p>
  *
  * <p><strong>Log level hierarchy</strong></p>
  * <ol>
- *     <li>{@link ServiceLogger#asError() Error}</li>
- *     <li>{@link ServiceLogger#asWarning() Warning}</li>
- *     <li>{@link ServiceLogger#asInfo() Info}</li>
- *     <li>{@link ServiceLogger#asVerbose() Verbose}</li>
+ * <li>{@link ClientLogger#asError() Error}</li>
+ * <li>{@link ClientLogger#asWarning() Warning}</li>
+ * <li>{@link ClientLogger#asInfo() Info}</li>
+ * <li>{@link ClientLogger#asVerbose() Verbose}</li>
  * </ol>
  *
  * @see Configuration
  */
-public class ServiceLogger implements ServiceLoggerAPI {
-    private static final NoopServiceLogger NOOP_LOGGER = new NoopServiceLogger();
+public class ClientLogger {
+    private static final NoopClientLogger NOOP_LOGGER = new NoopClientLogger();
 
     private final Logger logger;
 
+    /**
+     * Indicate that log level is at trace level.
+     */
+    public static final int TRACE_LEVEL = 0;
+
+    /**
+     * Indicate that log level is at verbose level.
+     */
+    public static final int VERBOSE_LEVEL = 1;
+
+    /**
+     * Indicate that log level is at information level.
+     */
+    public static final int INFORMATIONAL_LEVEL = 2;
+
+    /**
+     * Indicate that log level is at warning level.
+     */
+    public static final int WARNING_LEVEL = 3;
+
+    /**
+     * Indicate that log level is at error level.
+     */
+    public static final int ERROR_LEVEL = 4;
+
+    /**
+     * Indicate that logging is disabled.
+     */
+    public static final int DISABLED_LEVEL = 5;
+
+    private static final int DEFAULT_LOG_LEVEL = INFORMATIONAL_LEVEL;
     private int level = DEFAULT_LOG_LEVEL;
 
     private int configurationLevel;
 
     /**
      * Retrieves a logger for the passed class using the {@link LoggerFactory}.
+     *
      * @param clazz Class creating the logger.
      */
-    public ServiceLogger(Class clazz) {
+    public ClientLogger(Class clazz) {
         this(clazz.getName());
     }
 
     /**
      * Retrieves a logger for the passed class name using the {@link LoggerFactory}.
+     *
      * @param className Class name creating the logger.
      */
-    public ServiceLogger(String className) {
+    public ClientLogger(String className) {
         logger = LoggerFactory.getLogger(className);
     }
 
     /**
      * Sets the logger to the verbose logging level.
-     * @return Updated ServiceLogger if debug is enabled, otherwise a no-op logger.
+     *
+     * @return Updated ClientLogger if debug is enabled, otherwise a no-op logger.
```

Comment:
This is on me due to the initial implementation, mind changing this to:

@return the updated ClientLogger if debug is enabled, otherwise a no-op logger.

We may want to remove the bit about debug is enabled and about the no-op logger if we determine if the no-op logger is need at all.

Reply:
removing  no-op logger.